packages/x-data-grid-pro/src/tests/columns.DataGridPro.test.tsx

```diff
@@ -615,6 +615,30 @@ describe('<DataGridPro /> - Columns', () => {
       });
     });
 
+    it('should work with custom column header sort icon', async () => {
+      const iconSize = 24;
+      const iconScrollSize = iconSize + 1;
+      const gap = 2;
+      const padding = 20;
+      function CustomSortIcon() {
+        return <span style={{ width: iconSize, flex: 'none' }}>⇅</span>;
+      }
+      render(
+        <Test
+          rows={rows}
+          columns={columns}
+          slots={{
+            columnHeaderSortIcon: CustomSortIcon,
+          }}
+        />,
+      );
+      await act(async () => apiRef.current?.autosizeColumns());
+      expect(getWidths()).to.deep.equal([
+        padding + 132 + gap + iconScrollSize,
```

Comment:
Where do these values come from? Could you add a comment at least to explain them?